clang/include/clang/Driver/Options.td

```diff
@@ -2900,7 +2900,7 @@ defm clangir : BoolFOption<"clangir",
   PosFlag<SetTrue, [], [ClangOption, CC1Option], "Use the ClangIR pipeline to compile">,
   NegFlag<SetFalse, [], [ClangOption, CC1Option], "Use the AST -> LLVM pipeline to compile">,
   BothFlags<[], [ClangOption, CC1Option], "">>;
-def emit_cir : Flag<["-"], "emit-cir">, Visibility<[CC1Option]>,
+def emit_cir : Flag<["-"], "emit-cir">, Visibility<[ClangOption, CC1Option]>,
```

Comment:
I had to readd this. The code in `lib/Driver/Driver.cpp:4810`ish needs to know the file output type from flags to the driver. So we need **a driver flag**. And if we need a driver flag then we might as well merge it with one that correctly feeds the emission type along as well.

Reply:
Hmmm, this is content for Aaron, who was opposed to my take.